photon-client/src/main/scala/com/linkedin/photon/ml/data/DataValidators.scala

```diff
@@ -157,15 +158,17 @@ object DataValidators extends Logging {
    * @param labeledPoint The input data point
    * @return Whether the offset of the input data point is finite
    */
-  def finiteOffset(labeledPoint: LabeledPoint): Boolean = !(labeledPoint.offset.isNaN || labeledPoint.offset.isInfinite)
+  def finiteOffset(labeledPoint: LabeledPoint): Boolean =
+    !(labeledPoint.offset.isNaN || labeledPoint.offset.isInfinite)
 
   /**
    * Verify that a data point has a finite weight.
    *
    * @param labeledPoint The input data point
    * @return Whether the weight of the input data point is finite
    */
-  def finiteWeight(labeledPoint: LabeledPoint): Boolean = !(labeledPoint.weight.isNaN || labeledPoint.weight.isInfinite)
+  def nonNegativeWeight(labeledPoint: LabeledPoint): Boolean =
+    !(labeledPoint.weight.isNaN || labeledPoint.weight.isInfinite) && (labeledPoint.weight > 0)
```

Comment:
Here we're comparing with 0, but below in `rowHasNegativeWeight`, we compare with `MathConst.EPSILON`. Should be consistent.